web/server/codechecker_server/api/mass_store_run.py

```diff
@@ -948,6 +949,53 @@ def get_missing_file_ids(report: Report) -> List[str]:
 
         return True
 
+    def __check_report_count(self, report_dir: Path):
```

Comment:
Isn't at least part of this function redundant with the solution seen in #4018? These patches try to achieve very similar things. Can't we somehow merge these algorithms?

Reply:
This is called on the server, the other is on the client. After both patches are merged, we can consolidate the implementations somewhere.